httpbakery/request.go

```diff
@@ -159,7 +159,8 @@ func (r *readStopper) Close() error {
 	return r.rreq.close()
 }
 
-var nopCloserType = reflect.TypeOf(ioutil.NopCloser(nil))
+var nopCloserType = reflect.TypeOf(io.NopCloser(nil))
+var nopCloserWriterToType = reflect.TypeOf(io.NopCloser(bytes.NewReader([]byte{})))
```

Comment:
Just for reference, I don't love the approach we're using here (see comments [here](https://github.com/go-macaroon-bakery/macaroon-bakery/issues/276)), but it seems a reasonable and minimally-invasive compromise.